pkg/parcacol/querier.go

```diff
@@ -897,6 +908,24 @@ func (q *Querier) SymbolizeArrowRecord(
 		}
 		valueColumn := r.Column(indices[0]).(*array.Int64)
 
+		var valuePerSecondColumn arrow.Array
+		if isDelta(meta.PeriodType) {
```

Comment:
Do you mean, that instead of checking it here like this, pass in a `isDelta bool` into the SymbolizeArrow function, like we pass the valueColumnName?

Reply:
I don't know off the top of my head where `SymbolizeArrowRecord` is called, but yes we can better determine whether we're looking at a delta profile using the query instead of the PeriodType.